build-info-extractor/src/main/java/org/jfrog/build/extractor/usageReport/UsageReporter.java

```diff
@@ -1,40 +1,63 @@
 package org.jfrog.build.extractor.usageReport;
 
+import org.apache.commons.codec.digest.DigestUtils;
 import org.jfrog.build.api.util.Log;
 import org.jfrog.build.client.ProxyConfiguration;
 import org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager;
 
 import java.io.IOException;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
 
 /**
  * Created by Bar Belity on 20/11/2019.
  */
 public class UsageReporter {
-    private String productId;
+    private final String productId;
     private FeatureId[] features;
+    private String uniqueClientId;
 
     public UsageReporter(String productId, String[] featureIds) {
         this.productId = productId;
         setFeatures(featureIds);
     }
 
+    private String generateUniqueClientId() throws SocketException {
+        byte[] macAddress = getMacAddress();
+        return macAddress != null ? DigestUtils.sha1Hex(macAddress) : null;
+    }
+
+    @SuppressWarnings("unused")
     public void reportUsage(String artifactoryUrl, String username, String password, String accessToken, ProxyConfiguration proxyConfiguration, Log log) throws IOException {
         try (ArtifactoryManager artifactoryManager = new ArtifactoryManager(artifactoryUrl, username, password, accessToken, log)) {
             if (proxyConfiguration != null) {
                 artifactoryManager.setProxyConfiguration(proxyConfiguration);
             }
+            try {
+                uniqueClientId = uniqueClientId == null ? generateUniqueClientId() : uniqueClientId;
+            } catch (SocketException e) {
+                log.debug("Wasn't able to generate unique client ID.");
```

Comment:
Exceptions mustn't be ignored
```suggestion
                log.warn("Wasn't able to generate unique client ID: " + ExceptionUtils.getRootCauseMessage(e));
```